rolling-shutter/keyper/keypermetrics/metrics.go

```diff
@@ -107,69 +125,93 @@ func InitMetrics(dbpool *pgxpool.Pool, config kprconfig.Config) {
-		log.Error().Err(err).Msg("keypermetrics | Failed to get dkg result")
-		return
-	}
-	if dkgResult.Success {
-		MetricsKeyperDKGstatus.WithLabelValues(strconv.FormatInt(eons[len(eons)-1].Eon, 10)).Set(1)
+		log.Error().Err(err).Msg("keypermetrics | Failed to fetch batch configs")
 	} else {
-		MetricsKeyperDKGstatus.WithLabelValues(strconv.FormatInt(eons[len(eons)-1].Eon, 10)).Set(0)
+		currentAddress := config.GetAddress().Hex()
+
+		for _, batchConfig := range batchConfigs {
+			batchConfigIndexStr := strconv.Itoa(int(batchConfig.KeyperConfigIndex))
+
+			// Join keyper addresses for the label
+			keyperAddresses := strings.Join(batchConfig.Keypers, ",")
+			MetricsKeyperBatchConfigInfo.WithLabelValues(batchConfigIndexStr, keyperAddresses).Set(1)
+
+			// Check if current node is a keyper in this batch config
+			isKeyper := false
+			for _, keyperAddr := range batchConfig.Keypers {
+				if strings.EqualFold(keyperAddr, currentAddress) {
+					isKeyper = true
+					break
+				}
+			}
+
+			var isKeyperValue float64
+			if isKeyper {
+				isKeyperValue = 1
+			}
+			MetricsKeyperIsKeyper.WithLabelValues(batchConfigIndexStr).Set(isKeyperValue)
+		}
 	}
 
-	version, err := chainsync.GetClientVersion(context.Background(), config.Ethereum.EthereumURL)
+	// Populate MetricsKeyperDKGStatus
+	dkgResults, err := queries.GetAllDKGResults(ctx)
 	if err != nil {
-		log.Error().Err(err).Msg("execution_client_version metrics | Failed to get execution client version")
-		return
+		log.Error().Err(err).Msg("keypermetrics | Failed to fetch DKG results")
+	} else {
+		dkgResultMap := make(map[int64]database.DkgResult)
+		for _, result := range dkgResults {
+			dkgResultMap[result.Eon] = result
+		}
+
+		// Set DKG status for all eons
+		for _, eon := range eons {
+			eonStr := strconv.FormatInt(eon.Eon, 10)
+
+			if dkgResult, exists := dkgResultMap[eon.Eon]; exists {
+				var dkgStatusValue float64
+				if dkgResult.Success {
+					dkgStatusValue = 1
+				}
+				MetricsKeyperDKGStatus.WithLabelValues(eonStr).Set(dkgStatusValue)
+			} else {
+				// No DKG result found for this eon, set to 0
+				MetricsKeyperDKGStatus.WithLabelValues(eonStr).Set(0)
+			}
+		}
 	}
-
-	executionClientVersion := prometheus.NewGauge(
-		prometheus.GaugeOpts{
-			Namespace: "shutter",
-			Subsystem: "keyper",
-			Name:      "execution_client_version",
-			Help:      "Version of the execution client",
-			ConstLabels: prometheus.Labels{
-				"version": version,
-			},
-		},
-	)
-	executionClientVersion.Set(1)
-
-	prometheus.MustRegister(executionClientVersion)
-	metricsKeyperEthAddress := prometheus.NewGauge(
-		prometheus.GaugeOpts{
-			Namespace: "shutter",
-			Subsystem: "keyper",
-			Name:      "address",
-			Help:      "Ethereum address of the Keyper",
-			ConstLabels: prometheus.Labels{
-				"address": config.GetAddress().Hex(),
-			},
-		},
-	)
-	metricsKeyperEthAddress.Set(1)
-
-	prometheus.MustRegister(metricsKeyperEthAddress)
```

Comment:
Was there a specific need to change these to public and not keep using the static label version?

Reply:
Yeah it's fine. I was just wondering.